rs/replicated_state/src/canister_state/tests.rs

```diff
@@ -159,12 +162,32 @@ fn canister_state_push_input_request_success() {
             InputQueueType::RemoteSubnet,
         )
         .unwrap();
+    // Request was enqueued.
+    assert!(fixture.canister_state.has_input());
 }
 
 #[test]
-fn canister_state_push_input_response_no_reserved_slot() {
+fn canister_state_push_input_response_success() {
     let mut fixture = CanisterStateFixture::new();
-    let response = default_input_response(fixture.make_callback());
+    // Reserve a slot in the input queue.
+    fixture.with_input_slot_reservation();
+    // Pushing input response should succeed.
+    let response = default_input_response(fixture.make_callback(NO_DEADLINE), NO_DEADLINE).into();
```

Comment:
Do we want a case with a best effort message as well or would this not add value?

Reply:
I don't think it would add much. On the happy path, the difference between enqueuing a best-effort and a guaranteed response is whether or not we're also checking for a memory reservation. Plus we already have tests for that lower down in the call stack.